presto-spi/src/main/java/com/facebook/presto/spi/relation/LambdaDefinitionExpression.java

```diff
@@ -133,7 +133,7 @@ public CanonicalizeExpression(List<String> arguments, List<Type> argumentTypes)
         @Override
         public String visitCall(CallExpression call, Void context)
         {
-            return format("%s.%s(%s)", call.getFunctionHandle().getFunctionNamespace(), call.getDisplayName(), String.join(", ", call.getArguments().stream().map(e -> e.accept(this, null)).collect(Collectors.toList())));
+            return format("%s.%s(%s):%s", call.getFunctionHandle().getFunctionNamespace(), call.getDisplayName(), String.join(", ", call.getArguments().stream().map(e -> e.accept(this, null)).collect(Collectors.toList())), call.getType());
```

Comment:
Probably better to do it only for CAST so we don't accidentally cause other issues?

Reply:
For all other functions, return type should be the same if name and input type is the same. So this is just adding a little overhead. However, in any case that return type is different, it would be wrong to canonicalize them. So I think it's safe to keep it for all functions. Also not all cast functions are just called "cast". We don't really have an error prone way to identify all cast functions...

Reply:
LGTM

Reply:
probably good to have one parameter per line now? 😄